Convert class based to functional components

Rashmil-1999 opened the pull request:
This PR converts the Data Viewer and DFR3 Viewer components to Functional components from Class-based.

longshuicy commented:
@Vismayak could you take a first pass review of this PR? Since your keycloak PR might need this. Thank you so much!

Vismayak commented:
@Rashmil-1999  I get an error when I first go the DFR3Viewer, then to any other Viewer(Data, Hazard or Semantic Viewer) and back to the DFR3FViewer. This is the console error messages I am getting. Are you also getting the same error?
![image](https://github.com/IN-CORE/incore-ui/assets/16596733/f0473097-9246-45ea-896b-c1c00d50a4e0)

Rashmil-1999 commented:
So it seems like a side effect of using sync await. if you wait like 5 seconds before jumping to other component, everything works fine otherwise it basically mutates a state even though the component was unmounted... classbased didn't cause such issues but using hooks you've got to be more careful it seems :( anyways good catch! let me see how to fix this.

longshuicy commented on src/components/DFR3Viewer.jsx:
```diff
@@ -759,522 +505,286 @@ class DFR3Viewer extends React.Component {
+			if (is3dCurve(DFR3Curve)){
+				console.log(DFR3Curve.id)
+			}
+			curvesWithInfo.push(DFR3Curve);
```
Since you are checking on the fly, don't you want to remove that part?

Rashmil-1999 replied:
Ahh, yeah, I forgot lol I wanted to find out what curve was with 3d plot... Ill remove it. Thanks

longshuicy reviewed (changes requested):
Thank you for putting so much effort in refactoring this to hooks!
All the functionality works, just some minor comments. 

Do we have task for refactoring SemanticViewer, Profile, and hazardViewer? If not could you help create separate task so in the future we could distribute the work looking at your example.

Rashmil-1999 commented on src/components/DFR3Viewer.jsx:
```diff
@@ -1,8 +1,9 @@
+import { CopyToClipboard } from "react-copy-to-clipboard";
+import LoadingOverlay from "react-loading-overlay";
+
 // import ThreeDimensionalPlot from "./children/ThreeDimensionalPlot";
```
yep

Rashmil-1999 commented on src/components/DFR3Viewer.jsx:
```diff
@@ -113,555 +124,176 @@ const styles = {
+		)(dispatch);
+	}, [selectedDFR3Type, selectedInventory, selectedSpace, selectedHazard]);
+
+	const setCommonParameters = () => {
```
okay

Rashmil-1999 commented on src/components/DataViewer.jsx:
```diff
@@ -110,267 +122,180 @@ const styles = {
 		let url = `${config.dataService}/${datasetId}/blob`;
 
-		let response = await fetch(url, { method: "GET", mode: "cors", headers: await getHeader() });
+		let response = await fetch(url, { method: "GET", mode: "cors", headers: getHeader() });
```
ahh okay :)

ywkim312 commented:
The red boxed text should be aligned? If you compare with prod, you will probably see what I mean
![image](https://github.com/IN-CORE/incore-ui/assets/12547573/9ceffea9-8081-4739-9bee-fae4d705f115)

ywkim312 commented on src/components/CommunityApps.jsx:
```diff
@@ -0,0 +1,196 @@
+import React from "react";
```
how do I test this page? This is */playbook url right?

ywkim312 commented:
@Rashmil-1999 I briefly checked the code but I am not very good at react. So if you let me know what to checked in terms of functionality, I can perform that. I have deployed this to dev and checked some viewers and it looks okay

longshuicy reviewed (changes requested):
Code looks good. 

A few thing I noticed:

- [ ] Search seems weird. E.g. When search an item doesn't exist; I clear the search box then search again, the search action is actually never fired leaving the page stuck on empty results. 

<img width="1037" alt="image" src="https://github.com/IN-CORE/incore-ui/assets/13950475/798a953d-837e-4af0-adc2-896a54bd82e2">

----

- [ ] 3D fragility is not searchable and not showing up properly. e.g. search 6303c9e2cef2881193f226bd

<img width="1701" alt="image" src="https://github.com/IN-CORE/incore-ui/assets/13950475/fad73393-0064-44a7-b6da-4270b923c288">

---

- [ ] certain fragility breaks. e.g. 6489ba3355647c1f4968bbb8
<img width="639" alt="image" src="https://github.com/IN-CORE/incore-ui/assets/13950475/ac7f1c07-8a7a-4528-9ea7-9ba4ede9c0e1">

Rashmil-1999 commented:
> Code looks good.
> 
> A few thing I noticed:
> 
> * [x]  Search seems weird. E.g. When search an item doesn't exist; I clear the search box then search again, the search action is actually never fired leaving the page stuck on empty results.
> 
> <img alt="image" width="1037" src="https://private-user-images.githubusercontent.com/13950475/344315671-798a953d-837e-4af0-adc2-896a54bd82e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0NTM0OTIsIm5iZiI6MTcyMDQ1MzE5MiwicGF0aCI6Ii8xMzk1MDQ3NS8zNDQzMTU2NzEtNzk4YTk1M2QtODM3ZS00YWYwLWFkYzItODk2YTU0YmQ4MmUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA4VDE1Mzk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzZGY1ZDJlZWQ1ZTI2Mzk4ZjRhNDAwYjNjYWM4M2VmNzFjZjk2N2RmZGQ3NGZhZmZjMTc0MTNjOGZlYjVmNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7H6YmKIRUnYR11XaRUeMhmN856LpEa_JioHiWF42Y4w">
> * [ ]  3D fragility is not searchable and not showing up properly. e.g. search 6303c9e2cef2881193f226bd
> 
> <img alt="image" width="1701" src="https://private-user-images.githubusercontent.com/13950475/344317056-fad73393-0064-44a7-b6da-4270b923c288.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0NTM0OTIsIm5iZiI6MTcyMDQ1MzE5MiwicGF0aCI6Ii8xMzk1MDQ3NS8zNDQzMTcwNTYtZmFkNzMzOTMtMDA2NC00NGE3LWI2ZGEtNDI3MGI5MjNjMjg4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA4VDE1Mzk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2MzcwMWU4YTdkMzZhYzUxNzEwMjBkMzJkMzNjYzQyN2MxNDI2ODY1YzBmMDIxYjI3YTE0MjkyNzI1NGI2MjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DoXeXrMcQmNT86gt-4dC8Z1oBHh_6-LutYMht09ntrU">
> * [ ]  certain fragility breaks. e.g. 6489ba3355647c1f4968bbb8
> 
> <img alt="image" width="639" src="https://private-user-images.githubusercontent.com/13950475/344317764-ac7f1c07-8a7a-4528-9ea7-9ba4ede9c0e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0NTM0OTIsIm5iZiI6MTcyMDQ1MzE5MiwicGF0aCI6Ii8xMzk1MDQ3NS8zNDQzMTc3NjQtYWM3ZjFjMDctOGE3YS00NTI4LTllYTctOWJhNGVkZTljMGUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA4VDE1Mzk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM0YTRiY2YxYjlmYjIxMjViZTYxNTMwMThmMDAxMDNlMTY5YjRkZDYxNDNjMjU2MTg5MjIwMGFiZjY3MmNjNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mplTbkhpg4s63wRHZ0BUyR5jd5Nt_XZA4HxgZRbdgZc">

For the 1st one, everything seems to be working on my end I tried garbage val, got to the empty page and cleared the search bar and tried another string, it worked. Still I replaced the deprecated `onKeyPressed` with `onKeyDown` and it should work :/ could you please try this again.

For the 2nd and the 3rd one, I fixed the error, I have no clue how it is working in production lol but it is. I have replaced the check for is3dPlot property on the curve with is3dCurve function that we have defined. With this it seems to be showing up. For the search, I can search the ID's you gave me. Could you let me know what Issues you are facing related to search?

Rashmil-1999 commented on package-lock.json:
```diff
@@ -16,6 +16,7 @@
         "highcharts": "^9.1.0",
         "jstat": "1.5.3",
         "jwt-decode": "^3.1.2",
+        "keycloak-js": "^23.0.6",
```
apparently it is not there in the latest develop. It seems with some conflict resolution, this might have been missed. So should we let it be here?

longshuicy commented:
3d Fragility curves should have a different icon e.g. 6303c9e2cef2881193f226bd
<img width="414" alt="image" src="https://github.com/user-attachments/assets/9fb70503-c24b-4827-856d-6c16e3585e53">

<img width="1045" alt="image" src="https://github.com/user-attachments/assets/9d31f336-7d2a-404d-a06d-4592f0309fbc">

navarroc reviewed (approved):
Functionality of the updated components seems to be working as expected and matches the behavior of the latest on production so I have approved this change, but I have two general comments/questions:

1) There are a lot of warnings/errors about style in the console, should these be addressed at some point or the style/linting updated?

2) When selecting filters (e.g. in the DFR3 viewer), should they stay selected when doing a search? Production clears them out as well when entering text and hitting search so just asking what the better behavior is since I would think filters would stay and it would search the filtered results.
